### DISCUSSION TITLE
A CSP compliant petite-vue version

### DISCUSSION THREAD

**Discussion by baptistejamin** [vuejs/petite-vue#129] [Show and tell]
Hey guys!

Letting you know we are working on a CSP-compliant version of petite-vue.

I basically used the AngularJS 1 Lexer/Parser, and rewrote it in Vanilla and replaced the petite-vue Function evaluation with it.

I plan to release the AngularJS 1 Parser as a standalone NPM package

We already have a working prototype and we will show our fork very soon.

**Comment by baptistejamin** [comment]
Here is the forked version: https://github.com/baptistejamin/petite-vue-csp

Not available yet on NPM

**Comment by alshdavid** [comment]
`petite-vue` is exactly what we need in the front end world. A tiny, unopinionated template renderer that we can layer tools on top of as we feel we need. I can't thank you enough for this work.

- Want to use TypeScript? 
  - Great, use it - whatever version you want whenever you want.
- Prefer Rollup over Webpack? 
  - Great, no need for any plugins.
- Want to use a different approach to state management? 
  - No worries, pvue is just a template renderer - how the data gets there is not a concern to it.
- What about components!!?
  - Just wrap a pvue app in a W3C custom element   
- OMG I want to use EJS/Go templates/PHP!
  - Ok, sure - The client side view renderer doesn't care.

All I need from this is CSP compliance, dynamic template rendering, custom directives and the smallest file size humanly possible. I am looking forward to seeing how small (kb) it can become.

**Comment by baptistejamin** [comment]
The petite vue CSP version is around 30KB, witch is around the double of the regular build. At the end, it’s 12 KB once gzipped, witch is totally fine.

I already made my possible to reduce the size, but it’s not possible to continue in that direction without removing some features or safety.

To me 30KB is totally decent.
